src/libaktualizr/storage/sqlstorage.cc

```diff
@@ -1570,6 +1573,14 @@ void SQLStorage::saveReportEvent(const Json::Value& json_value) {
     LOG_ERROR << "Failed to save report event: " << db.errmsg();
     return;
   }
+
+  statement = db.prepareStatement<int>(
+      "DELETE FROM report_events WHERE id NOT IN (SELECT id FROM report_events ORDER BY id DESC LIMIT ?)",
```

Comment:
Maybe it makes sense to run the `COUNT()` at first since it is not "expensive" operation. Then only if the "count" is higher than `MAX_PENDING_EVENTS` to run the `delete in (select)` statement. Thus, we don't need to run the `delete and select` on each save which seems quite expensive operation. WDYT?